scripts/change_file_ext_test.py

```diff
@@ -0,0 +1,164 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/change_file_ext.py."""
+
+import os
+import shutil
+import tempfile
+
+from core.tests import test_utils
+from scripts import change_file_ext
+
+
+class ChangeExtensionsTest(test_utils.GenericTestBase):
+    """Test the change extension method."""
+
+    def setUp(self):
+        """Create dummy folders for testing and run the change extensions method
+        on the dummy files.
+        """
+        super(ChangeExtensionsTest, self).setUp()
+
+        self.rename_dir = tempfile.mkdtemp(suffix='dir_used_in_renaming')
+        self.non_rename_dir = tempfile.mkdtemp(
+            suffix='dir_not_used_in_renaming')
+
+        os.mkdir('%s/subdir' % self.rename_dir)
+        os.mkdir('%s/subdir' % self.non_rename_dir)
+
+        self.js_file_in_dir = 'js_file_in_dir.js'
+        self.js_file_in_subdir = 'js_file_in_subdir.js'
+        self.python_test_file = 'python_test_file.py'
+        self.ts_test_file = 'ts_test_file.ts'
+
+        self.js_file_in_dir_content = 'var a = 123;'
+        self.js_file_in_subdir_content = (
+            'function myFunction(p1, p2) {\n'
+            '  return p1 * p2;\n'
+            '}')
+        self.python_test_file_content = 'print \'I am test file!\''
+        self.ts_test_file_content = 'var a = 123;'
+
+        with open(
+            '%s/%s' % (self.rename_dir, self.js_file_in_dir), 'w') as f:
+            f.write(self.js_file_in_dir_content)
+
+        with open(
+            '%s/subdir/%s' % (
+                self.rename_dir, self.js_file_in_subdir), 'w') as f:
+            f.write(self.js_file_in_subdir_content)
+
+        with open('%s/%s' % (self.rename_dir, self.python_test_file), 'w') as f:
+            f.write(self.python_test_file_content)
+
+        with open('%s/%s' % (self.rename_dir, self.ts_test_file), 'w') as f:
+            f.write(self.ts_test_file_content)
+
+        with open(
+            '%s/%s' % (self.non_rename_dir, self.js_file_in_dir), 'w') as f:
+            f.write(self.js_file_in_dir_content)
+
+        with open(
+            '%s/subdir/%s' % (
+                self.non_rename_dir, self.js_file_in_subdir), 'w') as f:
+            f.write(self.js_file_in_subdir_content)
+
+        with open(
+            '%s/%s' % (self.non_rename_dir, self.python_test_file), 'w') as f:
+            f.write(self.python_test_file_content)
+
+        with open('%s/%s' % (self.non_rename_dir, self.ts_test_file), 'w') as f:
+            f.write(self.ts_test_file_content)
+
+        change_file_ext.change_extension(
```

Comment:
This seems like the actual test action, not setup. Probably should move this to the actual tests.

Reply:
I added two different test:
- First for ensuring that only the specified files are renamed
- Second for ensuring that file content remains unchanged

Both of these test require calling the `change_extension` method. So, I added it to setup itself. If we move it to test, we can go with two options:

- Merge the two test and call the function once initially
- Call the function independently in both test but rename files back to `.js` before calling the function for the seconf time.

I think it is better to go with the second one to have each test check exactly one behaviour.

What do you suggest?

Reply:
Hi @seanlip, may you please give a suggestion on this, Thanks!

Reply:
Sorry I missed this, thanks for the ping!

Yes, the second one is better. Can you check if the renaming back to .js is actually needed though? I thought setUp() gets run for each test (you can add an assertion before the renaming to verify that).

Reply:
> Sorry I missed this, thanks for the ping!

No issues!

> Yes, the second one is better. Can you check if the renaming back to .js is actually needed though? I thought setUp() gets run for each test (you can add an assertion before the renaming to verify that).

Ah yes, you are right setUp() runs for each test. I confirmed this using the following code in the second test:

```
        actual_files_in_dir_used_in_renaming = []
        for (dirpath, _, filenames) in os.walk(self.rename_dir):
            for filename in filenames:
                actual_files_in_dir_used_in_renaming.append(
                    os.path.join(dirpath, filename))
        expected_files_in_dir_used_in_renaming = [
            '%s/python_test_file.py' % self.rename_dir,
            '%s/js_file_in_dir.js' % self.rename_dir,
            '%s/ts_test_file.ts' % self.rename_dir,
            '%s/subdir/js_file_in_subdir.js' % self.rename_dir]
        self.assertEqual(set(actual_files_in_dir_used_in_renaming), 
            set(expected_files_in_dir_used_in_renaming))
```